sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/util/DateTimeUtilsSuite.scala

```diff
@@ -456,6 +456,18 @@ class DateTimeUtilsSuite extends SparkFunSuite with Matchers {
     }
   }
 
+  test("trailing colon while converting string to timestamp") {
+    val s = UTF8String.fromString("2019-10-31T10:59:23Z:::")
+    val time = DateTimeUtils.stringToTimestamp(s, defaultZoneId)
+    assert(time == None)
+  }
+
+  test("trailing space while converting string to timestamp") {
+    val s = UTF8String.fromString("2019-10-31T10:59:23ZZ   ")
```

Comment:
`stringToTimestamp` returns None becase of invalid time zone `ZZ` at the end. If you replace the string `"2019-10-31T10:59:23ZZ   "` by `"2019-10-31T10:59:23Z   "`, the functions shouldn't return None

Reply:
You are right this works accurately for trailing spaces, I first enounctered this issue on an older release branch and it did not have the `.trim` . But the fix should work for trailing spaces as well if backported.